packages/common/src/components/PkgGraph.tsx

```diff
@@ -0,0 +1,143 @@
+import { ReactWidget } from '@jupyterlab/apputils';
+
+import {
+  Graph,
+  GraphData,
+  GraphNode,
+  GraphLink,
+  GraphConfiguration
+} from 'react-d3-graph';
+
+import * as React from 'react';
+
+import { Conda } from '../tokens';
+
+/**
+ * Package graph property
+ */
+export interface IPkgGraphProps {
+  /**
+   * Package manager for the selected environment
+   */
+  pkgManager: Conda.IPackageManager;
+  /**
+   * Package name
+   */
+  package: string;
+  /**
+   * Graph configuration
+   */
+  config: GraphConfiguration<GraphNode, GraphLink> | Record<string, any>;
+}
+
+/**
+ * Package graph state
+ */
+export interface IPkgGraphState {
+  /**
+   * Graph data
+   */
+  data: GraphData<GraphNode, GraphLink> | null;
+}
+
+export class PkgGraph extends React.Component<IPkgGraphProps, IPkgGraphState> {
```

Comment:
Wondering whether React hooks could be used directly for new components, instead of class based components.

Since the current react in `peerDependencies` should support them (added in 16.8.0).

Reply:
(this could be part of a separate change)

Reply:
Yep, we can use hooks. I was just following the extension code style.

Reply:
We can open an issue to change all the React classes to hooks.